experimental/linting_tool/lib/model/profiles_store.dart

```diff
@@ -49,8 +49,12 @@ class ProfilesStore extends ChangeNotifier {
   }
 
   Future<void> addToExistingProfile(RulesProfile profile, Rule rule) async {
-    RulesProfile newProfile =
-        RulesProfile(name: profile.name, rules: profile.rules..add(rule));
+    var rules = profile.rules;
```

Comment:
Should rules be a `set` (assuming `rule` implements equality and hash and all that jazz)?

Reply:
The `Rule` class extends `Equatable`, which handles this for us(at least for now). Do you think that it might be a good idea to refactor this to a `Set` anyway?

Reply:
I'd use https://api.dartlang.org/stable/dart-collection/LinkedHashSet-class.html to get set semantics while retaining ordering.

Actually, it's probably worth thinking about using https://api.dartlang.org/stable/dart-collection/SplayTreeSet-class.html and order rules lexicographically

Reply:
Thanks a lot for this!
I have added a TODO for the time being to remind me of the refactor later on.